src/core/geocoding/qgsgooglemapsgeocoder.cpp

```diff
@@ -0,0 +1,253 @@
+  }
+  query.addQueryItem( QStringLiteral( "sensor" ), QStringLiteral( "false" ) );
+  query.addQueryItem( QStringLiteral( "address" ), address );
+  query.addQueryItem( QStringLiteral( "key" ), mApiKey );
+  res.setQuery( query );
+
+
+  if ( res.toString().contains( QLatin1String( "fake_qgis_http_endpoint" ) ) )
+  {
+    // Just for testing with local files instead of http:// resources
+    QString modifiedUrlString = res.toString();
+    // Qt5 does URL encoding from some reason (of the FILTER parameter for example)
+    modifiedUrlString = QUrl::fromPercentEncoding( modifiedUrlString.toUtf8() );
+    modifiedUrlString.replace( QLatin1String( "fake_qgis_http_endpoint/" ), QLatin1String( "fake_qgis_http_endpoint_" ) );
+    QgsDebugMsg( QStringLiteral( "Get %1" ).arg( modifiedUrlString ) );
+    modifiedUrlString = modifiedUrlString.mid( QStringLiteral( "http://" ).size() );
+    QString args = modifiedUrlString.mid( modifiedUrlString.indexOf( '?' ) );
+    if ( modifiedUrlString.size() > 150 )
+    {
+      args = QCryptographicHash::hash( args.toUtf8(), QCryptographicHash::Md5 ).toHex();
+    }
+    else
+    {
+      args.replace( QLatin1String( "?" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( "&" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( "<" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( ">" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( "'" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( "\"" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( " " ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( ":" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( "/" ), QLatin1String( "_" ) );
+      args.replace( QLatin1String( "\n" ), QLatin1String( "_" ) );
+    }
+#ifdef Q_OS_WIN
+    // Passing "urls" like "http://c:/path" to QUrl 'eats' the : after c,
+    // so we must restore it
+    if ( modifiedUrlString[1] == '/' )
+    {
+      modifiedUrlString = modifiedUrlString[0] + ":/" + modifiedUrlString.mid( 2 );
+    }
+#endif
+    modifiedUrlString = modifiedUrlString.mid( 0, modifiedUrlString.indexOf( '?' ) ) + args;
+    QgsDebugMsg( QStringLiteral( "Get %1 (after laundering)" ).arg( modifiedUrlString ) );
+    res = QUrl::fromLocalFile( modifiedUrlString );
+  }
+
+  return res;
+}
+
+QgsGeocoderResult QgsGoogleMapsGeocoder::jsonToResult( const QVariantMap &json ) const
+{
+  const QVariantMap geometry = json.value( QStringLiteral( "geometry" ) ).toMap();
+  const QVariantMap location = geometry.value( QStringLiteral( "location" ) ).toMap();
+  const double latitude = location.value( QStringLiteral( "lat" ) ).toDouble();
+  const double longitude = location.value( QStringLiteral( "lng" ) ).toDouble();
+
+  const QgsGeometry geom = QgsGeometry::fromPointXY( QgsPointXY( longitude, latitude ) );
+
+  QgsGeocoderResult res( json.value( QStringLiteral( "formatted_address" ) ).toString(),
```

Comment:
could we have a way to be a bit finer in the results and have a group and description properties as string? If given they would be used in the locator filter result.

Reply:
I've added group, and the api for description. There's nothing returned by the google maps geocoder which lends itself to the description field though (rather that applies more to the Google places api, which could be a potential other geocoder class!)

Reply:
perfect, thanks a lot!